### DISCUSSION TITLE
Changing fill or opacity of specific bar in RadialBar chart

### DISCUSSION THREAD

**Discussion by idjuradj** [recharts/recharts#3655] [Q&A] [answered]
Hi all, I am working on a RadialBar chart. One effect I would like to achieve is that when you click somewhere outside of the chart (custom legend component) bar that represents value of the clicked (selected) items gets highlighted in some way - ideally, fill opacity of other bars gets reduced, but some border on the selected bar would do as well.

Example:
Initial state of the chart
<img width="225" alt="Screenshot 2023-07-14 at 01 00 41" src="https://github.com/recharts/recharts/assets/5818649/891acc16-892d-407e-b518-8da652b209a6">

Then, when you click legend item for the greend circle (second looking from the outside inwards), just that green round keeps it's opacity, others fade to 10%.
<img width="204" alt="Screenshot 2023-07-14 at 01 02 39" src="https://github.com/recharts/recharts/assets/5818649/d60adc94-667f-4e92-a720-b4bf6e71b618">

Hope you understood what I meant. Thanks for the help in advance

**Comment by ckifer** [comment]
Hi! 

Could you give a codesandbox of the initial state of your chart so others can try to help?

**Comment by Yilun-Sun** [comment, marked as answer]
Hi @idjuradj , thanks for the question, that is an interesting one!

I believe this is exactly what you want -> [Codesandbox](https://codesandbox.io/s/simple-radial-bar-chart-forked-cqt639?file=/src/App.tsx)
<img width="518" alt="image" src="https://github.com/recharts/recharts/assets/43896664/9468d06a-f166-4a7d-a4dd-25bded04c7de">

The solution is very simple, use a state to keep the selected radial bar name, and use `Cell` as a child of `RadialBar` to get control of the selected radial bar opacity.
```tsx
<RadialBar
  background
  dataKey="uv"
>
  {data.map((entry, index) => (
    <Cell
      key={`cell-${index}`}
      opacity={selectedRadialBar === entry.name ? 1 : 0.1}
    />
  ))}
</RadialBar>
<Legend
  iconSize={10}
  width={120}
  height={140}
  layout="vertical"
  verticalAlign="middle"
  wrapperStyle={style}
  content={({ payload }) => (
    <ul>
      {payload?.map((entry, index) => {
        return (
          <li
            onClick={(data) => setSelectedRadialBar(entry.value)}
            key={`item-${index}`}
            style={{
              color: entry.color,
            }}
          >
            {entry.value}
          </li>
        );
      })}
    </ul>
  )}
/>
```

Also, if you want the legends get the related opacity, you can add `opacity: selectedRadialBar === entry.value ? 1 : 0.2` to the list item style.

**Reply by ckifer** [reply]
nice work @Yilun-Sun - we should add this to the storybook
